src/main/java/io/strimzi/kafka/quotas/VolumeUsageResult.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.quotas;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+
+/**
+ * <p>The result of an observation of the volumes in your kafka cluster.</p>
+ * The result can either be:
+ * <ul>
+ * <li><strong>successful</strong>; status is SUCCESS and volumeUsages is derived from a successful observation of the whole cluster</li>
+ * <li><strong>failed</strong>; status is not SUCCESS and volumeUsages is empty</li>
+ * </ul>
+ */
+public class VolumeUsageResult {
+
+    /**
+     * The observed usage on a per-volume basis.
+     * @return observed volume usage, this is empty if the status is not SUCCESS
+     */
+    public Collection<VolumeUsage> getVolumeUsages() {
+        return Collections.unmodifiableCollection(volumeUsages);
+    }
+
+    /**
+     * The status of this observation.
+     * @return categorises the result of the observation, either SUCCESS or another status that indicated failure
+     */
+    public VolumeSourceObservationStatus getStatus() {
+        return status;
+    }
+
+    /**
+     * The outcome of an observation
+     */
+    public enum VolumeSourceObservationStatus {
+        /**
+         * successfully observed the volumes of the cluster
+         */
+        SUCCESS,
+
+        /**
+         * timed out waiting on the future which is to be completed with the observed volumes from the cluster
+         */
+        SAFETY_TIMEOUT,
+
+        /**
+         * there was an error describing the cluster, its future completed exceptionally
+         */
+        DESCRIBE_CLUSTER_ERROR,
+
+        /**
+         * there was an error describing the log dirs, its future completed exceptionally
+         */
+        DESCRIBE_LOG_DIR_ERROR,
+
+        /**
+         * there was an unknown runtime exception while attempting to observe the cluster
+         */
+        EXCEPTION,
+
+        /**
+         * thread interrupted while trying to observe the cluster
+         */
+        INTERRUPTED,
+
+        /**
+         * execution exception while attempting to observe the cluster
+         */
+        EXECUTION_EXCEPTION
```

Comment:
`EXECUTION_EXCEPTION`, `SAFETY_TIMEOUT`, `INTERRUPTED` are all classifications for exceptions that can be thrown when waiting on the `Future` from the `run`. They do sound generic, which is misleading, maybe they should be combined into something like `VOLUME_USAGE_EXCEPTION` for any failure in the top `run` method, this might be more resistant to refactoring too.

The enum is looking forward to failure metrics. Having a stable classifications for each outcome means we can emit 0 immediately for all cases, heading off issues with alerting on `rate/increase` if we create the metric dynamically on first failure. Passing back the exception could lose context about which operation is failing. We also have a direct path from the metric name to an enum usage in code when trying to trace why it was emitted.

So it depends what we want in the metrics. If we want to have counters that break-down into different failure reasons then I'd prefer to have them enumerated up front. If we instead go for just success and fail count and use logs to debug why it's failing we don't need the enum.